### PR TITLE
sql: fill pg_extension.geometry_columns/geography_columns

### PR DESCRIPTION
**Pull request by otan**
Release note (sql change): Populate the pg_extension.geometry_columns
and pg_extension.geography_columns virtual table with metadata for
tables which store have columns relevant to geometry or geography data
types.

### CODE REVIEW

**Comment by rohany** [comment]
I'll take a closer look at this later, but the basic populate function is lazy -- you don't need a special generator

File: pkg/sql/pg_extension.go [hunk 1]
```diff
@@ -32,6 +36,64 @@ var pgExtension = virtualSchema{
...
+			hideVirtual,
+			func(db *sqlbase.DatabaseDescriptor, scName string, table *sqlbase.TableDescriptor) error {
+				if !table.IsPhysicalTable() {
+					return nil
+				}
+				for _, colDesc := range table.Columns {
+					if colDesc.Type.Family() != matchingFamily {
+						continue
+					}
+					m := colDesc.Type.InternalType.GeoMetadata
```
Comment (rohany) [inline pkg/sql/pg_extension.go:56]: I think generally we don't expose access to InternalType

File: pkg/sql/pg_extension.go [hunk 2]
```diff
@@ -32,6 +36,64 @@ var pgExtension = virtualSchema{
...
+							datumNDims = tree.DNull
+						}
+					}
+
+					shapeName := m.Shape.String()
+					if m.Shape == geopb.Shape_Unset {
+						shapeName = geopb.Shape_Geometry.String()
+					}
+
+					if err := addRow(
```
Comment (rohany) [inline pkg/sql/pg_extension.go:79]: does this table have oids?

File: pkg/sql/pg_extension.go [hunk 3]
```diff
@@ -32,6 +36,64 @@ var pgExtension = virtualSchema{
 	validWithNoDatabaseContext: false,
 }
 
+func postgisColumnsTablePopulator(
```
Comment (rohany) [inline pkg/sql/pg_extension.go:39]: The paradigm we have is to define the populate function normally, and call out to a function that takes in `addRow` instead -- I think its a bit cleaner that way.

File: pkg/sql/pg_extension.go [hunk 4]
```diff
@@ -44,9 +106,7 @@ CREATE TABLE pg_extension.geography_columns (
 	srid integer,
 	type text
 )`,
-	generator: func(ctx context.Context, p *planner, db *DatabaseDescriptor) (virtualTableGenerator, cleanupFunc, error) {
-		return nil, func() {}, errors.Newf("not yet implemented")
-	},
+	populate: postgisColumnsTablePopulator(types.GeographyFamily),
```
Comment (rohany) [inline pkg/sql/pg_extension.go:109]: Are there any permissions that we need to consider here?

Comment (otan) [reply pkg/sql/pg_extension.go:109]: i added one fix

File: pkg/sql/pg_extension.go (hunk 2 above)
Comment (otan) [reply pkg/sql/pg_extension.go:79]: nope.

File: pkg/sql/pg_extension.go (hunk 1 above)
Comment (otan) [reply pkg/sql/pg_extension.go:56]: fixed.

File: pkg/sql/pg_extension.go (hunk 3 above)
Comment (otan) [reply pkg/sql/pg_extension.go:39]: hmm, the problem then is i'd be duplicating the same code twice. seems cleaner to define this just once.

Comment (rohany) [reply pkg/sql/pg_extension.go:39]: Valid

**Review by rohany — APPROVED** [review]

**Comment by otan** [comment]
bors r=rohany

**Pull request merged**